### DISCUSSION TITLE
Add git autofetch option

### DISCUSSION THREAD

**Discussion by tzabbi** [zed-industries/zed#24611] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

Hey there,

I love the functionality of VSCode to fetch the latest changes from Git Remote at regular intervals. 

In VSCode there are the following settings for that function available:

```
  // When set to true, commits will automatically be fetched from the default remote of the current Git repository. Setting to `all` will fetch from all remotes.
  "git.autofetch": false,

  // Duration in seconds between each automatic git fetch, when `git.autofetch` is enabled.
  "git.autofetchPeriod": 180, 
 ```
 
 See: https://code.visualstudio.com/docs/sourcecontrol/overview#_remotes
 
 

### Environment

Ubuntu: 22.04.5
Zed: 0.162.5

**Comment by bonniesimon** [comment]
This would be an excellent addition. Waiting for this.

**Reply by eduardohki** [reply]
Just to give more context why this is important: when working with colleagues on the same repository using [Trunk-based Development](https://trunkbaseddevelopment.com/5-min-overview/), it is crucial to have the local copy of the `trunk` branch (probably `main` for this matter) always in sync with the origin. Not having to do it manually is great 😃

**Reply by boabdilperez** [reply]
I agree. My colleagues and collaborate by using individual branches off of a common feature branch and I would love to automatically update from upstream. Forgetting to do this leads to git conflicts which this would help greatly with!

**Comment by unisol1020** [comment]
+1

**Comment by andriygm** [comment]
relevant PR here https://github.com/zed-industries/zed/pull/53089
